src/main/java/org/spongepowered/common/mixin/api/mcp/entity/passive/CatEntityMixin_API.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.api.mcp.entity.passive;
+
+import net.minecraft.entity.ai.goal.TemptGoal;
+import net.minecraft.entity.monster.PhantomEntity;
+import net.minecraft.entity.passive.CatEntity;
+import net.minecraft.util.EntityPredicates;
+import org.spongepowered.api.data.value.Value;
+import org.spongepowered.api.entity.living.animal.Cat;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.util.Constants;
+
+import java.util.List;
+import java.util.Set;
+
+@Mixin(CatEntity.class)
+public abstract class CatEntityMixin_API extends TameableEntityMixin_API implements Cat {
+
+    @Shadow private TemptGoal temptGoal;
+    @Shadow protected abstract boolean shadow$func_213416_eg();
+    @Shadow protected abstract boolean shadow$func_213409_eh();
+
+    @Override
+    public boolean isPurring() {
```

Comment:
Why aren't these keys?

Reply:
`isPurring`, `isBeggingForFood`, `isHissing`

Reply:
@Cybermaxke Good question. I didn't make stuff like this keys because it didn't feel like data. Basically it doesn't correlate to any value off hand. It is a logic method.

Reply:
This is a impl detail, from API standpoint what says that it's actually a logic method?